multiversx_sdk/core/transactions_factories/smart_contract_transactions_factory.py

```diff
@@ -88,26 +87,31 @@ def create_transaction_for_execute(self,
                                        arguments: Sequence[Any] = [],
                                        native_transfer_amount: int = 0,
                                        token_transfers: Sequence[ITokenTransfer] = []) -> Transaction:
-        number_of_tokens = len(token_transfers)
+        transfers = list(token_transfers)
+        number_of_tokens = len(transfers)
         receiver = contract
 
         if native_transfer_amount and number_of_tokens:
-            raise BadUsageError("Can't send both native token and custom tokens(ESDT/NFT)")
+            native_tranfer = TokenTransfer.new_from_native_amount(native_transfer_amount)
+            transfers.append(native_tranfer)
```

Comment:
Optional, refactoring idea (without the need of an additional `transfers`):

```
token_transfers = token_transfers + [TokenTransfer.new_from_native_amount(native_transfer_amount)]
```

However, this works only if `+` is supported for `Sequence`, as well.

Reply:
I did it this way because by using `sequence` we make it immutable. If we modify the `token_tranfers` array and some one calls the function twice we may get unwanted results.

e.g: You send someone some esdts and also some native tokens. After that, maybe you want to send the same esdts to someone else but without the native tokens, so you would use the same list of esdts but don't provide the native amount. If in the method we modify the list, when calling the method the second time, you'll also have the native token transfer in the list.

Reply:
Generally speaking, the following creates a new sequence (thus, the input sequence is not mutated):

```
token_transfers = ... + ...
```